internal/manifests/collector/suite_test.go

```diff
@@ -15,170 +15,31 @@
 package collector
 
 import (
-	"context"
-	"crypto/tls"
 	"fmt"
-	"net"
 	"os"
-	"path/filepath"
-	"sync"
-	"testing"
-	"time"
 
-	routev1 "github.com/openshift/api/route/v1"
 	v1 "k8s.io/api/core/v1"
-	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"k8s.io/apimachinery/pkg/util/uuid"
-	"k8s.io/apimachinery/pkg/util/wait"
-	"k8s.io/client-go/kubernetes/scheme"
-	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/record"
-	"k8s.io/client-go/util/retry"
-	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/envtest"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
-	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
-	"sigs.k8s.io/controller-runtime/pkg/webhook"
 
 	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
 	"github.com/open-telemetry/opentelemetry-operator/internal/config"
 	"github.com/open-telemetry/opentelemetry-operator/internal/manifests"
-	"github.com/open-telemetry/opentelemetry-operator/internal/manifests/collector/testdata"
 )
 
 var (
-	k8sClient  client.Client
-	testEnv    *envtest.Environment
-	testScheme *runtime.Scheme = scheme.Scheme
-	ctx        context.Context
-	cancel     context.CancelFunc
-
-	logger = logf.Log.WithName("unit-tests")
-
+	logger      = logf.Log.WithName("unit-tests")
 	instanceUID = uuid.NewUUID()
-	err         error
-	cfg         *rest.Config
 )
 
 const (
 	defaultCollectorImage    = "default-collector"
 	defaultTaAllocationImage = "default-ta-allocator"
 )
 
-func TestMain(m *testing.M) {
```

Comment:
this was actually not needed at all and was slowing down tests

Reply:
How is the reconcile loop then tested?

Reply:
entirely within the new reconcile_test. This block was inside of the collector's manifest which is all essentially statically generated